modules/index-service/src/main/java/org/opencastproject/index/service/impl/IndexServiceImpl.java

```diff
@@ -760,13 +758,10 @@ private String startAddAssetWorkflow(JSONObject metadataJson, MediaPackage media
 
     try {
       // 1. Check if any active workflows are running for this mediapackage id
-      WorkflowSet workflowSet  = workflowService.getWorkflowInstances(new WorkflowQuery().withMediaPackage(mpId));
-      for (WorkflowInstance wf : Arrays.asList(workflowSet.getItems())) {
-        if (wf.isActive()) {
-          logger.warn("Unable to start new workflow '{}' on archived media package '{}', existing workfow {} is running",
-                  workflowDefId, mediaPackage, wf.getId());
-          throw new IllegalArgumentException("A workflow is already active for mp " + mpId + ", cannot start this workflow.");
-        }
+      if (workflowService.mediaPackageHasActiveWorkflows(mpId)) {
+        logger.warn("Unable to start new workflow '{}' on archived media package '{}', existing workflow is running",
+                workflowDefId, mediaPackage);
+        throw new IllegalArgumentException("A workflow is already active for mp " + mpId + ", cannot start this workflow.");
```

Comment:
No need for the warning if we then throw a similar error:

```suggestion
        throw new IllegalArgumentException("Unable to start new workflow '" + workflowDefId + "' on archived media package '" + mpId + "', existing workflow is running");
```